Update model to read in GRIB datasets

linjonathan opened the pull request:
Update model to read in GRIB datasets and use nanosecond precision for the time, suppressing warnings on nanosecond precision. This is necessary since there was an update to how netcdfs are served via the CDS platform.

Adds a new option to the namelist:
`file_type`

which can be set to either `netcdf` or `grib` depending on the file type of your data source. This should be backwards compatible with users who have already downloaded the netcdf-compatible data files of ERA5 from the CDS platform, though they will have issues when trying to download more years.

One key change in the namelist is the change of the `lvl` parameter to `isobaricInhPa`, which is the GRIB standard for isobaric pressure levels.

Commit:
update model to read in GRIB datasets and use nanosecond precision for time

linjonathan committed:
Update environment.yml

Add eccodes and cfgrib to environt

linjonathan committed:
Update README.md

Add eccodes and cfgrib to conda environment

linjonathan committed:
Update environment.yml

Commit:
include variable drag coefficient

priyanko97 commented:
I encountered an error while attempting to run python3 scripts/download_era5.py.

![Screenshot 2024-12-27 132631](https://github.com/user-attachments/assets/8d9f9f7a-6a68-4aab-9a20-da85c0dabd7f)

Commit:
replace Cd in init_m function

linjonathan committed:
include variable drag coefficient (#13)

Adds functionality for variable drag coefficient, which increases the
decay rate over land. Fixes a previous issue where storms did not decay
quickly enough over land, leading to positive biases in return period
calculations for inland locations.

Commit:
optimize parallel computation, fixed GRIB script

linjonathan committed:
Update compute.py

linjonathan reviewed (commented):
Looks good!
